### PR TITLE
Es6 lib geolocation

### PR DESCRIPTION
**Pull request by Calum-Campbell**
## What does this change?
 - Converts geolocation module to  ES6. 
## What is the value of this and can you measure success?
ES6 
## Does this affect other platforms - Amp, Apps, etc?
Shouldn't do
## Screenshots
No
## Tested in CODE?
No

### CODE REVIEW

**Review by nicl — APPROVED** [review]
👍 looks amazing to my untrained eyes (but probably check with a true client-sider too!)

File: static/src/javascripts/lib/geolocation.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    AU: 'AU',
+};
+const daysBeforeGeolocationRefresh = 10;
+
+const get = (): Promise<string> =>
+    new Promise((resolve, reject) => {
+        if (location) return resolve(location);
+
+        fetch(`${config.page.ajaxUrl}/geolocation`, {
+            method: 'GET',
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: We don't need this, because it's the default.

File: static/src/javascripts/lib/geolocation.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            contentType: 'application/json',
+            crossOrigin: true,
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: `crossOrigin: true,` -> `mode: 'cors'`

File: static/src/javascripts/lib/geolocation.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    US: 'US',
+    AU: 'AU',
+};
+const daysBeforeGeolocationRefresh = 10;
+
+const get = (): Promise<string> =>
+    new Promise((resolve, reject) => {
+        if (location) return resolve(location);
+
+        fetch(`${config.page.ajaxUrl}/geolocation`, {
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: Why not using `fetch-json` instead?

File: static/src/javascripts/lib/geolocation.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    'VA',
+    'AX',
+];
+
+// Returns one of { GB, US, AU, CA, EU, INT }
+// These are the different 'regions' we accept when taking payment.
+// See https://membership.theguardian.com/uk/supporter# for more context.
+const getSupporterPaymentRegion = (): string => {
+    const supporterLocation = getSync();
+    if (regionCountryCodes.indexOf(location) > -1) {
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: `regionCountryCodes.includes(location)`

File: static/src/javascripts/lib/geolocation.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        return supporterLocation;
+    }
+    if (europeCountryCodes.indexOf(location) > -1) {
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: `europeCountryCodes.includes(location)`

File: static/src/javascripts/lib/geolocation.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        return 'EU';
+    }
+    return 'INT';
+};
+
+const isInEurope = (): boolean => {
+    const countryCode = getSync();
+    return europeCountryCodes.indexOf(countryCode) > -1 || countryCode === 'GB';
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: `europeCountryCodes.includes(countryCode)`

File: static/src/javascripts/lib/geolocation.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+};
+
+export default {
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: default exports are considered to be harmful, according to @sndrs and @SiAdcock. Could you transform those into named exports please?

File: static/src/javascripts/lib/geolocation.js (hunk 3 above)
Comment (Calum-Campbell) [inline static/src/javascripts/lib/geolocation.js]: @gustavpursche Do you mean in terms of just naming?

File: static/src/javascripts/lib/geolocation.js (hunk 1 above)
Comment (Calum-Campbell) [inline static/src/javascripts/lib/geolocation.js]: Nice 👍

File: static/src/javascripts/lib/geolocation.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+];
+
+// Returns one of { GB, US, AU, CA, EU, INT }
+// These are the different 'regions' we accept when taking payment.
+// See https://membership.theguardian.com/uk/supporter# for more context.
+const getSupporterPaymentRegion = (): string => {
+    const supporterLocation = getSync();
+    if (regionCountryCodes.includes(location)) {
+        return supporterLocation;
+    }
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: Could we write it that way?

```
if (regionCountryCodes.includes(location)) {
    return getSync();
}
```

and kill `const supporterLocation = getSync();` then?

File: static/src/javascripts/lib/geolocation.js [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+        storage.local.set(storageKey, geolocation, {
+            expires: currentDate.setDate(
+                currentDate.getDate() + daysBeforeGeolocationRefresh
+            ),
+        });
+    });
+};
+
+const editionToGeolocation = (editionKey: string): string =>
+    editionToGeolocationMap[editionKey] || 'GB';
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: Better as:

```
const editionToGeolocation = (editionKey: string = 'GB'): string => editionToGeolocationMap[editionKey];
```

File: static/src/javascripts/lib/geolocation.js [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+const get = (): Promise<string> =>
+    new Promise((resolve, reject) => {
+        if (location) return resolve(location);
+
+        fetch('/geolocation', {
+            method: 'GET',
+            contentType: 'application/json',
+            mode: 'cors',
+        })
```
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: If we have to keep the `Content-Type` header, this has to be written as:

```
fetchJSON('/geolocation', {
    headers: {
        'Content-Type': 'application/json',
    }
    mode: 'cors',
})
```

Could you also rename `fetch` to `fetchJSON` to make it clear, we aren't using fetch here?

File: static/src/javascripts/lib/geolocation.js (hunk 8 above)
Comment (Calum-Campbell) [inline static/src/javascripts/lib/geolocation.js]: Yep good shout.

File: static/src/javascripts/lib/geolocation.js (hunk 9 above)
Comment (Calum-Campbell) [inline static/src/javascripts/lib/geolocation.js]: Also 👍

File: static/src/javascripts/lib/geolocation.js (hunk 10 above)
Comment (Calum-Campbell) [inline static/src/javascripts/lib/geolocation.js]: 👍

File: static/src/javascripts/lib/geolocation.js (hunk 9 above)
Comment (gustavpursche) [inline static/src/javascripts/lib/geolocation.js]: Oh, I made as mistake. The default paramter should be `UK` instead of `GB` ...

Comment (Calum-Campbell) [inline static/src/javascripts/lib/geolocation.js]: ah yes, good spot

**Review by gu-stav — APPROVED** [review]
Looks great now! 🌷 Thanks for addressing all the issues.

**Pull request merged**